yt_dlp/YoutubeDL.py

```diff
@@ -1379,6 +1379,14 @@ def _prepare_filename(self, info_dict, *, outtmpl=None, tmpl_type=None):
             if not filename:
                 return None
 
+            trim_file_name = self.params.get('trim_file_name')
+            if trim_file_name:
+                # https://github.com/yt-dlp/yt-dlp/issues/5526#issuecomment-1312783517
+                no_ext, *ext = filename.rsplit('.', info_dict.get('ext', '').count('.') + 1)
+                # cut filename and remove trailing spaces and extra dots
+                name = no_ext[:trim_file_name].strip().rstrip('.')
+                filename = join_nonempty(name, *ext, delim='.')
```

Comment:
Why strip trailing spaces and dots? Then we would be trimming beyond what the user specified

Reply:
Excuse my xonsh
```xonsh
 D:/ @ paths = [p" test ", p" . test . ", p" . test . .mp4"]
 D:/ @ for path in paths:
......     path.touch()
......     name, actual = map(repr, map(str, [path, *pg`*test*`]))
......     print(f"{name:<20} | {actual:<20}")
......     path.unlink()
......
' test '             | ' test'
' . test'            | ' . test'
' . test . .mp4'     | ' . test . .mp4'
```
As we can see, trailing dots and spaces are stripped on the OS Level and thats only if its actually trailing. Extensions save us from this problem, which means we wouldn't need to strip it at all.

While I can see why you wouldn't want starting spaces, this behavior is correct and required to be kept for compatibility.

Reply:
Also, trimming filenames will never *add* whitespace to the beginning of a filename, so stripping whitespace from the beginning should be of no concern to the trim filename code